src/main/java/frc/robot/commands/TwoBallAutoStraight.java

```diff
@@ -0,0 +1,87 @@
+package frc.robot.commands;
+
+import java.util.List;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.RamseteController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.math.trajectory.constraint.DifferentialDriveVoltageConstraint;
+import edu.wpi.first.wpilibj2.command.ParallelDeadlineGroup;
+import edu.wpi.first.wpilibj2.command.RamseteCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants;
+import frc.robot.FieldInformation;
+import frc.robot.subsystems.DriveTrain;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.Vision;
+
+public class TwoBallAutoStraight extends SequentialCommandGroup implements AutoCommandInterface {
+    static final double DISTANCE_BACK = 1.4;
+    Trajectory m_trajectory;
+    public TwoBallAutoStraight(Shooter shooter, Intake intake, DriveTrain driveTrain, Vision vision) {
+        var autoVoltageConstraint =
+        new DifferentialDriveVoltageConstraint(
+            new SimpleMotorFeedforward(Constants.ksVolts,
+                                       Constants.kvVoltSecondsPerMeter,
+                                       Constants.kaVoltSecondsSquaredPerMeter),
+            Constants.kDriveKinematics,
+            10);
+
+        TrajectoryConfig config =
+            new TrajectoryConfig(Constants.kMaxSpeed,
+                                Constants.kMaxAcceleration)
+                .setKinematics(Constants.kDriveKinematics)
+                .addConstraint(autoVoltageConstraint)
+                .setReversed(true);
+
+        Pose2d initialPose = getInitialPose();
+        Pose2d finalPose = new Pose2d(
```

Comment:
If you use the FieldInformation to get the location of the ball, we can use the same command for all TwoBallAuto commands.

Reply:
I'm not sure what you mean?